backup/backup.go

```diff
@@ -194,89 +202,87 @@ func (bm *BackupManager) Backup(ctx context.Context) error {
 	return nil
 }
 
-func (bm *BackupManager) ChoosePod(ctx context.Context, pods []*corev1.Pod) (int, error) {
+func (bm *BackupManager) GetUUIDSet(ctx context.Context, pods []*corev1.Pod) (map[string]string, error) {
 	cluster := bm.cluster
-	// if this is the first time
-	if cluster.Status.Backup.Time.IsZero() {
-		if len(pods) == 1 {
-			return 0, nil
-		}
+	uuids := make(map[string]string, len(pods))
+	for i := range pods {
+		if podIsReady(pods[i]) {
+			op, err := newOperator(cluster.PodHostname(i),
+				constants.MySQLPort,
+				constants.BackupUser,
+				bm.mysqlPassword,
+				bm.threads)
+			if err != nil {
+				return nil, fmt.Errorf("failed to create operator: %w", err)
+			}
+			defer op.Close()
 
-		for i := range pods {
-			if i == int(cluster.Status.CurrentPrimaryIndex) {
+			if err := op.GetServerStatus(ctx, &bm.status); err != nil {
 				continue
 			}
-			if podIsReady(pods[i]) {
-				return i, nil
-			}
+			uuids[strconv.Itoa(i)] = bm.status.UUID
 		}
-		return int(cluster.Status.CurrentPrimaryIndex), nil
-	}
-
-	lastIndex := cluster.Status.Backup.SourceIndex
-	op, err := newOperator(cluster.PodHostname(lastIndex),
-		constants.MySQLPort,
-		constants.BackupUser,
-		bm.mysqlPassword,
-		bm.threads)
-	if err != nil {
-		return -1, err
-	}
-	defer op.Close()
-
-	st := &bkop.ServerStatus{}
-	if err := op.GetServerStatus(ctx, st); err != nil {
-		return -1, err
 	}
+	return uuids, nil
+}
 
-	if st.UUID != cluster.Status.Backup.SourceUUID {
-		bm.log.Info("server_uuid of the last backup source has changed", "index", lastIndex)
+// ChoosePod chooses a pod to take a backup from.
+// It returns the index of the chosen pod and whether the backupBinlog should be skipped.
+func (bm *BackupManager) ChoosePod(ctx context.Context, pods []*corev1.Pod) (int, bool, error) {
```

Comment:
This function does not return error values.  All error values returned are `nil`
When all `pods` are not ready,  this function returns the PrimaryIndex as index of the selected pod.
It might be good to return an error indicating that there are no pods to backup.

Reply:
[8a56b75](https://github.com/cybozu-go/moco/pull/604/commits/8a56b759a2de0459f9c992b96b99f4e7f93149b3)